### ISSUE TITLE
Import marked as not accessed when symbol is aliased in inner scope

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1983]
Pyright contains special logic to handle the case where a symbol that is used within a class body is not assigned before being used but is assigned within an outer scope, such as the module-level (global) scope. However, this logic doesn't properly mark the outer symbol as accessed.

In the example below, the imported symbol `field` is reported as "not accessed".

```python
from dataclasses import dataclass, field

@dataclass
class Foo:
    field: str
    value: int = field(default=1)
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.149, which I just published. It will also be in the next release of pylance.
